services/apps/src/apps/routes/admin.clj

```diff
@@ -153,6 +156,58 @@
           on the fields included in the request."
           (ok (apps/admin-update-category current-user (assoc body :id category-id)))))
 
+(defroutes* admin-ontologies
+
+  (GET* "/" []
+        :query [params SecuredQueryParams]
+        :return ActiveOntologyDetailsList
+        :summary "List Ontology Details"
+        :description
+"Lists Ontology details saved in the metadata service.
+
+#### Delegates to metadata service
+    GET /ontologies"
+        (ok (admin/list-ontologies current-user)))
+
+  (POST* "/:ontology-version" []
+         :path-params [ontology-version :- OntologyVersionParam]
+         :query [params SecuredQueryParams]
+         :return AppCategoryOntologyVersionDetails
+         :summary "Set Active Ontology Version"
+         :description
+         "Sets the active `ontology-version` to use in non-admin endpoints when querying the ontology
+          endpoints of the metadata service."
+         (ok (admin/set-category-ontology-version current-user ontology-version)))
+
+  (GET* "/:ontology-version/:root-iri" []
+        :path-params [ontology-version :- OntologyVersionParam
+                      root-iri :- OntologyClassIRIParam]
+        :query [params SecuredQueryParams]
+        :summary "Get App Category Hierarchy"
+        :description
+"Gets the list of app categories that are visible to the user for the given `ontology-version`,
+ rooted at the given `root-iri`.
+
+#### Delegates to metadata service
+    POST /ontologies/{ontology-version}/{root-iri}/filter
+Please see the metadata service documentation for response information."
+        (listings/get-app-hierarchy current-user ontology-version root-iri))
+
+  (GET* "/:ontology-version/:root-iri/unclassified" [root-iri]
+        :path-params [ontology-version :- OntologyVersionParam
+                      root-iri :- OntologyClassIRIParam]
+        :query [params AppListingPagingParams]
+        :return AppListing
+        :summary "List Unclassified Apps"
+        :description
+"Lists all of the apps that are visible to the user that are not under the given `root-iri`, or any of
+ its subcategories, for the given `ontology-version`.
+
+#### Delegates to metadata service
+    POST /ontologies/{ontology-version}/{root-iri}/filter-unclassified"
+        (ok (coerce! AppListing
```

Comment:
I guess this is consistent with other parts of apps, but I believe this can just be done with the `:coercion` key on the route as long as `apps.util.coercions/custom-coercion-matcher` is made public (perhaps it belongs in common-swagger-api?), which might be nice to move toward rather than using the full reimplementations of `coerce!` etc. from that namespace. Not completely related to the actual work at hand though, just food for thought.